### PR TITLE
Fixed test-resources.json for Tables.

### PR DESCRIPTION
**Pull request by vcolin7**
Live tests were failing with [this error message](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1561772&view=logs&j=a7294988-27ec-5a5d-711d-c1eb36e49d88&t=daf21ffc-750b-51c3-5225-a3487c798ca6&l=1442). This PR adds the missing variable.

### CODE REVIEW

File: sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TableServiceClientTest.java [hunk 1]
```diff
@@ -83,19 +83,23 @@ public void serviceCreateTable() {
...
+
         // Arrange
         String tableName = testResourceNamer.randomName("tableName", 20);
 
         // The tenant ID does not matter as the correct on will be extracted from the authentication challenge in
         // contained in the response the server provides to a first "naive" unauthenticated request.
         final ClientSecretCredential credential = new ClientSecretCredentialBuilder()
-            .clientId(Configuration.getGlobalConfiguration().get("AZURE_TABLES_CLIENT_ID", "clientId"))
-            .clientSecret(Configuration.getGlobalConfiguration().get("AZURE_TABLES_CLIENT_SECRET", "clientSecret"))
+            .clientId(Configuration.getGlobalConfiguration().get("TABLES_CLIENT_ID", "clientId"))
```
[the remark this answers is not in the document]
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TableServiceClientTest.java:96]: They're created as environment variables by default by the New-TestResources PowerShell script.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java [hunk 2]
```diff
@@ -576,7 +576,9 @@ public TableClientBuilder tableName(String tableName) {
     }
 
     /**
-     * Enabled tenant discovery when authenticating with the Storage Table Service. This is disabled by default.
+     * Enable tenant discovery when authenticating with the Table Service. <strong>This functionality is disabled by
+     * default and only available for Storage endpoints using service version
+     * {@link TableServiceVersion#V2020_12_06 2020_12_06}.</strong>
```
[the remark this answers is not in the document]
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java:581]: After a chat with @srnagar we decided to leave it to the service to respond with an error. We have documented as best as we can and we prefer not to do this type of validations on the client side.

**Review by srnagar — APPROVED** [review]

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (christothes) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java:581]: My main concern was that the error that you get is simply that no challenge is issued, and the option seems not to do anything. The option, in isolation, is guaranteed to fail.

**Pull request merged**